src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -565,9 +565,14 @@ private static Assembly SgenAssemblyResolver(object source, ResolveEventArgs e)
                     return null;
                 }
 
-                if(s_referencedic.ContainsKey(assemblyname))
+                if (s_referencedic.ContainsKey(assemblyname))
                 {
                     string reference = s_referencedic[assemblyname];
+
+                    //for System.ServiceModel.Primitives, we need to load its execution assembly rather than reference assembly
+                    if (assemblyname.Equals("System.ServiceModel.Primitives"))
```

Comment:
What if there is third party assembly having the same name as System.ServiceModel.Primitives?

Reply:
I don't think this is a concern as it's understood that only framework libraries start with System. Even Microsoft will often release packages beginning with "Microsoft." to indicate that while it's created by Microsoft, it's not a core system library. Basically it's a reserved namespace by well understood convention. We only stuck with System for WCF because of the pre-existence of the code on .NET Framework and it would end up being a rename. Otherwise we likely wouldn't have used System as a prefix.

---
In reply to: [219902748](https://github.com/dotnet/corefx/pull/32403#discussion_r219902748) [](ancestors = 219902748)